### ISSUE TITLE
[Question] Computing normalization stats for RGB image datasets (CIFAR10, etc.)

### ISSUE THREAD

**Issue by pi-tau** [phlippe/uvadlc_notebooks#106]
**Tutorial**: 5

**Describe the bug**
In tutorial 5, Cell 4 we compute the mean and std of the CIFAR10 training dataset. Usually what I have seen people do is to compute the mean and std across all the channels together, but I see here that we are actually computing means and stds for each color channel separately. I guess that makes sense, because in most datasets you actually have very different scales for different features, and you actually normalize across every feature separately. I was wondering, 

Is it sensible to say that each of the color channels is a separate feature and are just people being lazy when they compute mean and std over all channels together? I was also wondering if you have seen this topic of pre-processing image data discussed somewhere formally (e.g. a paper or a book) ?

Also, in the text cell after that you explain `RandomResizedCrop` as re-scaling the image and cropping afterwards, but I think that it is actually a crop followed by a re-scale to the requested size (see here: https://pytorch.org/vision/0.8/_modules/torchvision/transforms/functional.html#resized_crop).

Thanks

**Comment by phlippe** [comment]
Hi @pi-tau, the normalization during preprocessing is used to have an input distribution with mean 0 and std 1. This is important, since we design our initialization (see Tutorial 4) and our activation functions (see Tutorial 3) with such a distribution in mind. There are several papers/blog posts on that, you can probably find them with searching for 'neural network input normalization'. 

In our CNNs, the first layer is a convolution, which inherently uses different weights for different channels. Therefore, it is recommended to normalize each channel independently. For example, this was commonly done for CNNs on the ImageNet dataset. However, in most image datasets, the statistics of the three channels are very similar. Thus, you will likely not see much of a difference in terms of performance, and with layers like BatchNorm, the network can correct for eventual biases. Note also that we do not have to normalize each input pixel (over height and width) independently, since the same weights of the convolution are applied on all pixels. 

Thanks for pointing out the flip of resizing and cropping, I adjusted the corresponding explanations in 529e1bea00bfc8167dbced09957b2324610568ac.

**Comment by pi-tau** [comment]
Hi,
thanks a lot for the thorough answer !

I have been also meaning to ask if there were any ideas about normalizing across `H` and `W` or maybe across some local patches, but I see now why nobody does that.
